src/mapblock_mesh.cpp

```diff
@@ -1215,7 +1216,7 @@ MapBlockMesh::MapBlockMesh(MeshMakeData *data, v3s16 camera_offset):
 		// Create meshbuffer
 		// This is a "Standard MeshBuffer",
 		// it's a typedeffed CMeshBuffer<video::S3DVertex>
-		scene::SMeshBuffer *buf = new scene::SMeshBuffer();
+		scene::SMeshBufferTangents *buf = new scene::SMeshBufferTangents();
```

Comment:
I dont think that different vertex types slows anything down. the only difference between those two types are that S3DVertexTangents holds 2 extra v3f - tangent and binormal. If shaders are used those vectors are passed to them.
Also, before we were calculating tangents in shaders, runtime, for each and every vertex.
Now those two are calculated only on mesh update, and thats major speedup.